tests/e2e/framework/testInfo.go

```diff
@@ -111,6 +134,87 @@ func (t testInfo) Update(r int) error {
 	return t.createStatusFile(r)
 }
 
+func (t testInfo) FetchAndSaveClusterLogs() error {
+	// initialize projectID if user did not specify
+	if *projectID == "" {
+		pjID, err := util.Shell("getProjectID.sh")
```

Comment:
Sorry If I confused you. If projectID is not set, you should just skip this (ie not do anything). You don't need tests/e2e/framework/getProjectID.sh script either. You might want to set it in Jenkinsfile when you call e2e.sh in https://github.com/istio/istio/blob/master/Jenkinsfile#L51. You can use env.PROJECT inside Jenkinsfile to find the project.